lib/resque/plugins/timed_round_robin/timed_round_robin.rb

```diff
@@ -6,22 +6,27 @@ def filter_busy_queues(qs)
     end
 
     def rotated_queues
-      # only refresh queues when a slice expires
-      @rtrr_queues ||= queues
-      if @rtrr_queues.empty?
-        @rtrr_queues = nil
-        return []
-      end
+      @rtrr_queues = fetch_queues
+      return [] if @rtrr_queues.empty?
 
       @n ||= 0
-      if slice_expired?
-        @rtrr_queues = queues
-        advance_offset
-      end
+      advance_offset if slice_expired?
 
       @rtrr_queues.rotate(@n)
     end
 
+    def fetch_queues
+      return @rtrr_queues unless @rtrr_queues&.empty? || queue_list_expired?
+      # refresh queues at a given interval, default is 60 seconds
+      @queue_list_expiration = Time.now + queue_refresh_interval
+      queues
+    end
+
+    def queue_list_expired?
+      @queue_list_expiration ||= Time.now
+      Time.now > @queue_list_expiration
```

Comment:
If `@queue_list_expiration == nil` at call-time, this will almost always but not always return `false` (unless nearby a millisecond boundary). Is that the desired behavior?

Reply:
Yep! This should only be nil when a new worker is created and at that point we want to be fetching a new list of queues.